sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -258,6 +302,29 @@ def decode_from_stream(self, in_stream, nested):
     return self._construct_from_components(
         [c.decode_from_stream(in_stream, True) for c in self._coder_impls])
 
+  def estimate_size(self, value):
+    """Estimates the encoded size of the given value, in bytes."""
+    estimated_size, _ = (
+        self.get_estimated_size_and_observables(value))
+    return estimated_size
+
+  def get_estimated_size_and_observables(self, value):
+    """Returns estimated size of value along with any nested observables."""
+    values = self._extract_components(value)
+    estimated_size = 0
+    observables = []
+    for i in range(0, len(self._coder_impls)):
+      child_value = values[i]
+      if isinstance(child_value, observable.ObservableMixin):
+        observables.append((child_value, self._coder_impls[i]))
+      else:
+        c = self._coder_impls[i]  # type cast
+        child_size, child_observables = (
+            c.get_estimated_size_and_observables(child_value))
+        estimated_size += child_size + ESTIMATED_NESTED_OVERHEAD
```

Comment:
It seems that get_estimated_size should take a nested parameter (or maybe assume nested) rather than assuming there's a fixed nesting overhead for the child coder. (E.g. this will be wrong if the components are ints or floats or tuples or lists or...)